### PR TITLE
min/max filters for monetary and frequency

### PR DESCRIPTION
**Pull request by mayurkmmt**
min/max filters for monetary and frequency in rfm

## Summary by CodeRabbit

- **New Features**
  - Added support for filtering customers by minimum and maximum monetary value and purchase frequency before segmentation.

- **Bug Fixes**
  - Improved error handling for invalid or inconsistent filter values.

- **Tests**
  - Expanded test coverage to validate filtering functionality, error handling, backward compatibility, lazy evaluation, caching, and edge cases.

### CODE REVIEW

File: tests/segmentation/test_rfm.py [hunk 1]
```diff
@@ -308,31 +302,163 @@ def test_single_bin_segments(self, larger_df):
...
-            f_segments=[1.0],
-            m_segments=[0.0, 0.5],
+            min_frequency=min_frequency,
+            max_frequency=max_frequency,
+        )
+        result_range = rfm_range.df
+        assert len(result_range) == expected_result_len
+        assert min_frequency <= result_range["frequency"].min() <= result_range["frequency"].max() <= max_frequency
+
+    @pytest.mark.parametrize(
```
[the remark this answers is not in the document]
Comment (mayurkmmt) [reply tests/segmentation/test_rfm.py:343]: Thanks @mvanwyk !!
